Changed fill.py example to be less misleading

samvaughan opened the pull request:
## PR Summary
I've changed /examples/lines_bars_and_markers/fill.py to be less misleading, following issues #5827 and #7956. 

These plots were over-simplified because fill was acting just like fill_between, since the final line of the region to be filled was completely along the x-axis. I've changed the first example to be a simple polygon (as suggested in #5827) and the second to not have the final point at y=0 anymore. I've also outlined the region being filled, to make what's going on more obvious. 

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

<!--
Thank you so much for your PR!  To help us review your contribution, please
consider the following points:

- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.

- Help with git and github is available at
  https://matplotlib.org/devel/gitwash/development_workflow.html.

- Do not create the PR out of master, but out of a separate branch.

- The PR title should summarize the changes, for example "Raise ValueError on
  non-numeric input to set_xlim".  Avoid non-descriptive titles such as
  "Addresses issue #8576".

- The summary should provide at least 1-2 sentences describing the pull request
  in detail (Why is this change required?  What problem does it solve?) and
  link to any relevant issues.

- If you are contributing fixes to docstrings, please pay attention to
  http://matplotlib.org/devel/documenting_mpl.html#formatting.  In particular,
  note the difference between using single backquotes, double backquotes, and
  asterisks in the markup.

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

timhoffm commented:
Thanks for the update.

https://8797-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/gallery/lines_bars_and_markers/fill.html?highlight=fill

I don't find the polygon particularly instructive:
- Start and endpoint are still at the same y.
- I think we could leave out the grid and thus the zorder.

maybe simply

~~~
x = [0, 1, 3, 1]
y = [1, 2, 1, 0]

fig, ax = plt.subplots()
ax.fill(x, y)
~~~

Minor things:
- Single blank lines in the code example are sufficient. No need for two blank lines.
- For completeness, the first plot should get a `plt.plot()` as well.

samvaughan commented:
Sure, no problem. I've removed the grid and made the first example a square
```
x = [0, 1, 2, 1]
y = [1, 2, 1, 0]
 
fig, ax = plt.subplots()
ax.fill(x, y)
```
and outlined it in black with `plt.plot()`. I've removed the double spaces too.

samvaughan commented:
No worries, just changed it!
